File: 20180716/Non Decresing Array.py
class Solution(object):
    def checkPossibility(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        ans = None
        for i in range(1, len(nums)):
            if nums[i] < nums[i-1]:
                if ans != None:
                    return False
                ans = i - 1
        return ans == 0 or ans == len(nums) - 2 or ans== None or nums[ans - 1] <= nums[ans + 1] or nums[ans] <= nums[ans + 2] 

File: 20180716/Non Decresing Array.java
class Solution {
    public boolean checkPossibility(int[] nums) {
        int ans = -1;
        for(int i = 1; i < nums.length; i++){
            if(nums[i] < nums[i-1]){
                if( ans != -1){
                    return false;
                }
                else{
                    ans = i - 1;
                }
            }
        }
        return ans == 0 || ans == nums.length - 2 || ans == -1 || nums[ans - 1] <= nums[ans + 1] || nums[ans] <= nums[ans + 2];
    }
}
